benefits/enrollment_switchio/api.py

```diff
@@ -133,25 +165,113 @@ def get_registration_status(self, registration_id, timeout=5) -> RegistrationSta
-        """
-        Creates named (on-disk) temp files for client cert auth.
-        * request_func: curried callable from `requests` library (e.g. `requests.get`).
-        """
-        # requests library reads temp files from file path
-        # The "with" context destroys temp files when response comes back
-        with NamedTemporaryFile("w+") as cert, NamedTemporaryFile("w+") as key, NamedTemporaryFile("w+") as ca:
-            # write client cert data to temp files
-            # resetting so they can be read again by requests
-            cert.write(self.client_certificate)
-            cert.seek(0)
 
-            key.write(self.private_key)
-            key.seek(0)
+@dataclass
+class Group:
+    id: int
+    operatorId: int
+    name: str
+    code: str
+    value: int
 
-            ca.write(self.ca_certificate)
-            ca.seek(0)
 
-            # request using temp file paths
-            return request_func(verify=ca.name, cert=(cert.name, key.name))
+class EnrollmentClient(Client):
+
+    def __init__(self, api_url, authorization_header_value, private_key, client_certificate, ca_certificate):
+        super().__init__(private_key, client_certificate, ca_certificate)
+        self.api_url = api_url
+        self.authorization_header_value = authorization_header_value
+
+    def _get_headers(self):
+        return {"Authorization": self.authorization_header_value}
+
+    def healthcheck(self, timeout=5):
+        request_path = "/api/external/discount/echo"
+
+        response = self._cert_request(
+            lambda verify, cert: requests.get(
+                self.api_url.strip("/") + request_path,
+                headers=self._get_headers(),
+                cert=cert,
+                verify=verify,
+                timeout=timeout,
+            )
+        )
+
+        response.raise_for_status()
+
+        return response.text
+
+    def get_groups(self, pto_id, timeout=5):
+        request_path = f"/api/external/discount/{pto_id}/groups"
+
+        response = self._cert_request(
+            lambda verify, cert: requests.get(
+                self.api_url.strip("/") + request_path,
+                headers=self._get_headers(),
+                cert=cert,
+                verify=verify,
+                timeout=timeout,
+            )
+        )
+
+        response.raise_for_status()
+
+        return [Group(**discount_group) for discount_group in response.json()]
+
+    def get_groups_for_token(self, pto_id, token, timeout=5):
+        request_path = f"/api/external/discount/{pto_id}/token/{token}"
+
+        response = self._cert_request(
+            lambda verify, cert: requests.get(
+                self.api_url.strip("/") + request_path,
+                headers=self._get_headers(),
+                cert=cert,
+                verify=verify,
+                timeout=timeout,
+            )
+        )
+
+        response.raise_for_status()
+
+        return response.json()
```

Comment:
Is this not a `Group` or list of them as well?

Reply:
It's a list of JSON objects containing just a `Group` identifier and expiration datetime, if applicable.

```
[
   {
      "group": "student-discount",
      "expiresAt": "2025-12-31T23:59:59"
   },
   {
      "group": "veteran-discount",
      "expiresAt": null
   }
]
```

We could introduce a dataclass for this. Any suggestions on the name?

Reply:
`GroupExpiry` sounds good. Added it in https://github.com/cal-itp/benefits/pull/2988/commits/a9ba91439b3f67796edd0103b34061fb12f3b798